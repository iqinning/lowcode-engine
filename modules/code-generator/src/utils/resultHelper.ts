import { ResultFile, ResultDir } from '@alilc/lowcode-types';

import { CodeGeneratorError } from '../types/error';
import { FlattenFile } from '../types/file';

export function createResultFile(name: string, ext = 'jsx', content = ''): ResultFile {
  return {
    name,
    ext,
    content,
  };
}

export function createResultDir(name: string): ResultDir {
  return {
    name,
    dirs: [],
    files: [],
  };
}

export function addDirectory(target: ResultDir, dir: ResultDir): void {
  if (target.dirs.findIndex((d) => d.name === dir.name) < 0) {
    target.dirs.push(dir);
  } else {
    throw new CodeGeneratorError(
      `Adding same directory to one directory: ${dir.name} -> ${target.name}`,
    );
  }
}

export function addFile(target: ResultDir, file: ResultFile): void {
  if (target.files.findIndex((f) => f.name === file.name && f.ext === file.ext) < 0) {
    target.files.push(file);
  } else {
    throw new CodeGeneratorError(
      `Adding same file to one directory: ${file.name} -> ${target.name}`,
    );
  }
}

export function flattenResult(dir: ResultDir, cwd = ''): FlattenFile[] {
  if (!dir.files.length && !dir.dirs.length) {
    return [];
  }

  return [
    ...dir.files.map(
      (file): FlattenFile => ({
        pathName: `${cwd ? `${cwd}/` : ''}${file.name}${file.ext ? `.${file.ext}` : ''}`,
        content: file.content,
      }),
    ),
  ].concat(
    ...dir.dirs.map((subDir) =>
      flattenResult(subDir, [cwd, subDir.name].filter((x) => x !== '' && x !== '.').join('/')),
    ),
  );
}
