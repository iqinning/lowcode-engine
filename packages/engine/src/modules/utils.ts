import { isFormEvent, compatibleLegaoSchema, getNodeSchemaById } from '@alilc/lowcode-utils';
import { isNodeSchema } from '@alilc/lowcode-types';

const utils = {
  isNodeSchema,
  isFormEvent,
  compatibleLegaoSchema,
  getNodeSchemaById,
};

export default utils;