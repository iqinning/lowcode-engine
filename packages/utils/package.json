{
  "name": "@alilc/lowcode-utils",
  "version": "1.0.0-beta.17",
  "description": "Utils for Ali lowCode engine",
  "files": [
    "lib",
    "es"
  ],
  "main": "lib/index.js",
  "module": "es/index.js",
  "scripts": {
    "build": "build-scripts build --skip-demo"
  },
  "dependencies": {
    "@alifd/next": "^1.19.16",
    "@alilc/lowcode-types": "1.0.0-beta.17",
    "lodash": "^4.17.21",
    "react": "^16",
    "zen-logger": "^1.1.0"
  },
  "devDependencies": {
    "@alib/build-scripts": "^0.1.18",
    "@types/node": "^13.7.1",
    "@types/react": "^16",
    "build-plugin-component": "^0.2.10"
  },
  "publishConfig": {
    "access": "public",
    "registry": "https://registry.npmjs.org/"
  },
  "gitHead": "7ceab30f7d55a4be02f766eccaa4b96d35804e0b"
}
