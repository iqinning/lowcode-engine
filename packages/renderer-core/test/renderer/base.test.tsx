import renderer from 'react-test-renderer';
import React from 'react';
import '../utils/react-env-init';
import pageRendererFactory from '../../src/renderer/renderer';
import { sampleSchema } from '../mock/sample';

describe('notFountComponent', () => {
  const Render = pageRendererFactory();

  const component = renderer.create(
    // @ts-ignore
    <Render
      schema={sampleSchema as any}
      components={{}}
      appHelper={{}}
    />,
  );

  it('not found snapshot', () => {
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
})