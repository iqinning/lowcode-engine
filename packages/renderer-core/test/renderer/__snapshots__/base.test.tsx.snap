// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`notFountComponent not found snapshot 1`] = `
<div
  className="lce-page _css_pseudo_node_ockyigdqxl1"
  style={Object {}}
>
  <div
    __componentName="RootHeader"
    __id="node_ockyigdqxl2"
  >
    Component Not Found
  </div>
  <div
    __componentName="RootContent"
    __id="node_ockyigdqxl3"
    contentBgColor="white"
    contentMargin="20"
    contentPadding="20"
  >
    <div
      __componentName="Button"
      __id="node_ockyigdqxl5"
      __style__={Object {}}
      baseIcon=""
      behavior="NORMAL"
      className="_css_pseudo_node_ockyigdqxl5"
      content="按 钮"
      events={
        Object {
          "ignored": true,
        }
      }
      fieldId="button_kyige3yf"
      loading={false}
      otherIcon=""
      size="medium"
      triggerEventsWhenLoading={false}
      type="primary"
    >
      Component Not Found
    </div>
  </div>
  <div
    __componentName="RootFooter"
    __id="node_ockyigdqxl4"
  >
    Component Not Found
  </div>
</div>
`;
